File: GaleriaArteJava/GaleriaArte/src/sor/galeriaarte/clases/Juddi.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package sor.galeriaarte.clases;

import org.uddi.api_v3.FindQualifiers;
import org.uddi.api_v3.FindService;
import org.uddi.api_v3.GetServiceDetail;
import org.uddi.api_v3.Name;
import org.uddi.api_v3.ServiceDetail;
import org.uddi.api_v3.ServiceInfo;
import org.uddi.api_v3.ServiceInfos;
import org.uddi.api_v3.ServiceList;
import org.uddi.v3_service.UDDIInquiryService;


/**
 *
 * @author ShadowLink
 */
public class Juddi {
    
    public String getServiceUrl(String name){
        ServiceInfo s = new ServiceInfo();
        UDDIInquiryService juddi = new UDDIInquiryService();
        FindService fs = new FindService();
        FindQualifiers fq = new FindQualifiers();
        fq.getFindQualifier().add("approximateMatch");
        Name n = new Name();
        n.setValue(name);
        n.setLang("en");
        fs.getName().add(n);
        fs.setFindQualifiers(fq);
        
        ServiceList services = juddi.getUDDIInquiryImplPort().findService(fs);
        ServiceInfos si = services.getServiceInfos();
        for(int i=0; i<si.getServiceInfo().size(); i++){
            s = si.getServiceInfo().get(i);
        }
        
        GetServiceDetail sd = new GetServiceDetail();
        sd.getServiceKey().add(s.getServiceKey());
        ServiceDetail service = juddi.getUDDIInquiryImplPort().getServiceDetail(sd);
        String url = service.getBusinessService().get(0).getBindingTemplates().getBindingTemplate().get(0).getAccessPoint().getValue();
        
        
        return url;
    }
}

File: GaleriaArteNet/GaleriaArte/Clases/Juddi.cs
/*
 * Created by SharpDevelop.
 * User: ShadowLink
 * Date: 05/03/2015
 * Time: 16:13
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using System.Diagnostics;
using System.Linq;
using System.Net;
using System.Windows.Forms.VisualStyles;

namespace GaleriaArte.Clases
{
	/// <summary>
	/// Description of Juddi.
	/// </summary>
	public class Juddi
	{
		public Juddi()
		{
			
		}
		
		public string getServiceUrl(string name){
			serviceInfo s = new serviceInfo();
        	UDDIInquiryService juddi = new UDDIInquiryService();
        	find_service fs = new find_service();
        	fs.findQualifiers = new []{"approximateMatch"};
        	name n = new name();
        	n.Value = name;
        	n.lang = "en";
        	fs.name = new []{n};
        	serviceList services = juddi.find_service(fs);
    
        	for(int i = 0; i<services.serviceInfos.Count(); i++){
        		s = services.serviceInfos[i];
        	}
        	
        	
        	get_serviceDetail sd= new get_serviceDetail();
        	sd.serviceKey = new string[]{s.serviceKey};
        	serviceDetail service = juddi.get_serviceDetail(sd);
        	string url = service.businessService[0].bindingTemplates[0].accessPoint.Value;
        	
        	if(checkUrl(url)){
        		url = url;
        	}else{
        		url =url;
        	}
        	
        	return url;
		}
		
		private bool checkUrl(string url){
			bool isAlive = true;
			WebRequest request = WebRequest.Create(url);
			HttpWebResponse response = (HttpWebResponse)request.GetResponse();
			if (response == null || response.StatusCode != HttpStatusCode.OK){
				isAlive = false;
			}
			response.Close();
			return isAlive;
		}
	}
}
